### ISSUE TITLE
Get us passing in CI again!

### ISSUE THREAD

**Issue by yaronyg** [thaliproject/Thali_CordovaPlugin#916]
Once we have #913 and #914 done and checked in then what we need to do is:

3. Merge vNext_yarong_417_812 with  #913/#914 CI and if they pass then we need to check them
1.1. We need to fix CI per the issue described below regarding coordinating across Android and iOS
1.2. Make sure UnitTest_App.js will run node tests (this really should have been fixed in vNext_yarong_417_812 but just in case be sure)
2. Re-apply shrinkwrap everywhere. @larryonoff made some package.json changes in his commit to vNext that when we sync with vNext is going to break shrinkwrap I think.
5. Then we can check in #840, #851 and #876 as they are reviewed and approved

**Comment by yaronyg** [comment]
Please wait until Andrew and I finish merging code in vNext_yarong_417_1

**Comment by artemjackson** [comment]
>  1. We need to cherry pick the updated jxcore-cordova bits from #889 and submit those to vNext

But jxcore-cordova that is used in #889(v0.1.2b) has been already merged into vNext (see #886)

**Comment by yaronyg** [comment]
The problem is that in step 1 above when we run the iOS and Android tests at the same time the Android and iOS devices will see each other because they are both using the wifi native mock and so all of their discovery announcements will be mutually viewable. 

We have two choices:
1. Run the Android and iOS tests serially, not in parallel.
2. Put the Android and iOS devices into a single pool and have the coordination server manage all the Android and iOS devices as one pool which are running the same tests at the same time.

I suspect that step 1 may be the least dangerous in the long run and require the smallest changes

**Comment by larryonoff** [comment]
We agreed with @yaronyg that option 3 will be implemented in the scope of this issue and option 2 will be implemented in the scope of another issue.

**Comment by yaronyg** [comment]
@artemjackson The 'service type' used in WiFi is set in ThaliWifiInfrastructure._shouldBeIgnored from the value thaliConfig.SSDP_NT. So if we edit thaliConfig.SSDP_NT to be say http://www.thaliproject.org/test/android for Android devices and http://www.thaliproject.org/test/ios for iOS devices then we can do 3 above and let the CI server run Android and iOS tests simultaneously without them interfering with each other.

**Comment by larryonoff** [comment]
@artemjackson please pick one of three modes and get it on CI working. Then open 2 more issues for two remained modes.

**Comment by artemjackson** [comment]
> We have two choices:
1. Run the Android and iOS tests serially, not in parallel.

@yaronyg It can be quickly implemented if add CLI arguments like `android-count` and `ios-count` to `runCoordinatedTests`, e.g. :
```
// run tests for 3 Android devices
runCoordinatedTests --android-count=3 

// run tests for 2 iOS devices
runCoordinatedTests --ios-count=2 

// run tests for 1 Android device and 2 iOS devices
runCoordinatedTests --android-count=1 --android-count=2 
```

**Comment by yaronyg** [comment]
@artemjackson Wouldn't this confuse the test runner? How will it know to report test result status in Github? I was really hoping we would just use the NT solution.

**Comment by artemjackson** [comment]
@yaronyg you are right, with current implementation of test runner it will work incorrectly

**Comment by larryonoff** [comment]
this issue is linked with #1029
